packages/firestore/test/unit/core/event_manager.test.ts

```diff
@@ -320,91 +320,6 @@ describe('QueryListener', () => {
   });
 
-  it('raises query metadata events only when hasPendingWrites on the query changes', () => {
-    const fullEvents: ViewSnapshot[] = [];
-    const query = Query.atPath(path('rooms'));
-    const doc1 = doc(
-      'rooms/Eros',
-      1,
-      { name: 'Eros' },
-      { hasLocalMutations: true }
-    );
-    const doc2 = doc(
-      'rooms/Hades',
-      2,
-      { name: 'Hades' },
-      { hasLocalMutations: true }
-    );
-    const doc1prime = doc('rooms/Eros', 1, { name: 'Eros' });
-    const doc2prime = doc('rooms/Hades', 2, { name: 'Hades' });
-    const doc3 = doc('rooms/Other', 3, { name: 'Other' });
-
-    const fullListener = queryListener(query, fullEvents, [], {
-      includeQueryMetadataChanges: true
-    });
-
-    const view = new View(query, documentKeySet());
-    const snap1 = applyDocChanges(view, doc1, doc2).snapshot!;
-    const snap2 = applyDocChanges(view, doc1prime).snapshot!;
-    const snap3 = applyDocChanges(view, doc3).snapshot!;
-    const snap4 = applyDocChanges(view, doc2prime).snapshot!;
-
-    fullListener.onViewSnapshot(snap1);
-    fullListener.onViewSnapshot(snap2); // Emits no events
-    fullListener.onViewSnapshot(snap3);
-    fullListener.onViewSnapshot(snap4); // Metadata change event.
-
-    const expectedSnap4 = {
-      query: snap4.query,
-      docs: snap4.docs,
-      oldDocs: snap3.docs,
-      docChanges: [],
-      fromCache: snap4.fromCache,
-      syncStateChanged: snap4.syncStateChanged,
-      hasPendingWrites: false
-    };
-    expect(fullEvents).to.deep.equal([snap1, snap3, expectedSnap4]);
-  });
-
-  it(
-    'Metadata-only document changes are filtered out when ' +
-      'includeDocumentMetadataChanges is false',
-    () => {
-      const filteredEvents: ViewSnapshot[] = [];
-      const query = Query.atPath(path('rooms'));
-      const doc1 = doc(
-        'rooms/Eros',
-        1,
-        { name: 'Eros' },
-        { hasLocalMutations: true }
-      );
-      const doc2 = doc('rooms/Hades', 2, { name: 'Hades' });
-      const doc1prime = doc('rooms/Eros', 1, { name: 'Eros' });
-      const doc3 = doc('rooms/Other', 3, { name: 'Other' });
-      const filteredListener = queryListener(query, filteredEvents);
-
-      const view = new View(query, documentKeySet());
-      const snap1 = applyDocChanges(view, doc1, doc2).snapshot!;
-      const snap2 = applyDocChanges(view, doc1prime, doc3).snapshot!;
-
-      const change3 = { type: ChangeType.Added, doc: doc3 };
-
-      filteredListener.onViewSnapshot(snap1);
-      filteredListener.onViewSnapshot(snap2);
-
-      const expectedSnap2 = {
-        query: snap2.query,
-        docs: snap2.docs,
-        oldDocs: snap1.docs,
-        docChanges: [change3],
-        fromCache: snap2.fromCache,
-        syncStateChanged: snap2.syncStateChanged,
-        hasPendingWrites: snap2.hasPendingWrites
-      };
-      expect(filteredEvents).to.deep.equal([snap1, expectedSnap2]);
-    }
-  );
```

Comment:
Android / iOS kept these tests...

Reply:
These tests require the internal distinction between document and query metadata changes. The only case that remains is tested in `raises metadata events only when specified`.